File: src/js/0002.js
var
  previous = 0,
  current = 1,
  sum = 0,
  temp;

while (current < 4e6) {
  if (current % 2 === 0) sum += current;
  temp = current;
  current = current + previous;
  previous = temp;
}

console.log(sum);

File: src/scala/0002.scala
val test = 4e6
var previous = 0
var current = 1
var temp = 0
var solution = 0

while (current < test) {
  if (current % 2 == 0) solution += current
  temp = current
  current = current + previous;
  previous = temp;
}
println(solution)
